Sources/SolToBoogie/ProcedureTranslator.cs

```diff
@@ -122,6 +122,40 @@ public override bool Visit(EnumDefinition node)
             return false;
         }
 
+        private BoogieCallCmd GenerateBoogieCallCmd(TypeDescription type, BoogieExpr value, string name)
+        {
+            if (type.TypeString.Equals("address") || type.TypeString.Equals("address payable") || type.IsDynamicArray() || type.IsStaticArray())
+            {
+                var callCmd = new BoogieCallCmd("boogie_si_record_sol2Bpl_ref", new List<BoogieExpr>() { value }, new List<BoogieIdentifierExpr>());
+                callCmd.Attributes = new List<BoogieAttribute>();
+                callCmd.Attributes.Add(new BoogieAttribute("cexpr", $"\"{name}\""));
+                return callCmd;

```

Comment:
I thought we wanted to refactor the logic that creates the callCmd by replacing the 3 lines with a function.